server.go

```diff
@@ -1394,10 +1402,55 @@ func (s *Server) Close() error {
 
 // closeListenerClients closes all clients on the specified listener.
 func (s *Server) closeListenerClients(listener string) {
+	// Get the clients associated with the specified listener.
 	clients := s.Clients.GetByListener(listener)
+	var wg sync.WaitGroup
+
+	// Create a channel to distribute clients among worker goroutines.
+	jobs := make(chan *Client, s.Options.Capabilities.ShutdownClientsWorkerNum)
+
+	// Create a context with a timeout. Adjust the timeout value as needed.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(s.Options.Capabilities.ShutdownClientsTimeout)*time.Second)
+	defer cancel()
+
+	// Start worker goroutines.
+	for i := 0; i < s.Options.Capabilities.ShutdownClientsWorkerNum; i++ {
+		wg.Add(1)
+		go func(clients <-chan *Client, ctx context.Context) {
```

Comment:
ctx should be first to keep standard context.Context pattern, unless this is a channel pattern that I am unaware of.